### DISCUSSION TITLE
Using TOC plugin from within a .md file?

### DISCUSSION THREAD

**Discussion by writeblankspace** [lumeland/lume#654] [Q&A]
I want to place a Table Of Contents within my `.md` file. This file is processed with `vto` and `md`.

However, `toc` seems to be undefined within it: 

```
> Cannot read properties of undefined (reading 'length')
```

Is there a way to access the TOC from within the `.md` file?

**Comment by oscarotero** [comment]
Can I see a code example to reproduce this error?

**Reply by writeblankspace** [reply]
`_components/toc.vto`:

```html
{{ if toc.length }}
<nav class="toc">
	<details>
		<summary>Table of contents</summary>
		<ul>
			{{ for header2 of toc }}
			<li>
				<a href="#{{ header2.slug }}">{{ header2.text }}</a>
				{{ if header2.children.length }}
				<ul>
					{{ for header3 of header2.children }}
					<li>
						<a href="#{{ header3.slug }}">{{ header3.text }}</a>
					</li>
					{{ /for }}
				</ul>
				{{ /if }}
			</li>
			{{ /for }}
		</ul>
	</details>
</nav>
{{ /if }}
```

`index.md`:

```md
---
templateEngine: [vto, md]
---

{{ comp.toc({ toc: toc }) }}
```

**Reply by oscarotero** [reply]
Thanks!

I see two things here:

- The template engine is configured to run `vto` first, so the `toc` is not yet created because it's created later by `md`.
- But changing the order (`md` first) doesn't work neither because the data object passed to `vto` is not the same object updated by `md`(with the new `toc` variable). And I'm not sure if it's easy to do that, will take a look.
